### PR TITLE
Add shared consts to oneOf objects

### PR DESCRIPTION
**Pull request by cjwooo**
## Description

Replaces https://github.com/faros-ai/airbyte-connectors/pull/865.
This PR makes the spec compatible for newer versions of Airbyte without changing the locations of anything in the resulting config object.

## Type of change
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

**Review by tovbinm — APPROVED** [review]

**Comment by thomas-gerber** [comment]
@cjwooo before you merge, a few questions:
1. [this](https://docs.airbyte.com/connector-development/connector-specification-reference/#using-oneof) is what you are trying to abide by, correct?
2. I do get the oneOf [here](https://github.com/faros-ai/airbyte-connectors/blob/25dbdb386183bf491915a7f8f000e69662c1a9ee/destinations/airbyte-faros-destination/resources/source-specific-configs/spec.json#L6) but I do not get [those](https://github.com/faros-ai/airbyte-connectors/blob/25dbdb386183bf491915a7f8f000e69662c1a9ee/destinations/airbyte-faros-destination/resources/source-specific-configs/agileaccelerator.json#L5); could you clarify?
3. in this iteration of the destination, is it still possible to have a single destination used by N different sources (i.e. fill several "source specific config" sections on the same destination)? Is it a problem for the current setup of CE?

**Review by ypc-faros — APPROVED** [review]

File: destinations/airbyte-faros-destination/resources/source-specific-configs/spec.json [hunk 1]
```diff
@@ -8,6 +8,10 @@
       "type": "object",
       "title": "Configurations",
       "properties": {
+        "option_title": {
```
[the remark this answers is not in the document]
Comment (cjwooo) [reply destinations/airbyte-faros-destination/resources/source-specific-configs/spec.json:11]: Every object defined in a `oneOf` has to have a `const` property, even if it's the only object, to be rendered properly in the Airbyte UI.

**Comment by tovbinm** [comment]
Thank you @cjwooo !!

**Comment by thomas-gerber** [comment]
> > @cjwooo before you merge, a few questions:
> > 
> > 1. [this](https://docs.airbyte.com/connector-development/connector-specification-reference/#using-oneof) is what you are trying to abide by, correct?
> > 2. I do get the oneOf [here](https://github.com/faros-ai/airbyte-connectors/blob/25dbdb386183bf491915a7f8f000e69662c1a9ee/destinations/airbyte-faros-destination/resources/source-specific-configs/spec.json#L6) but I do not get [those](https://github.com/faros-ai/airbyte-connectors/blob/25dbdb386183bf491915a7f8f000e69662c1a9ee/destinations/airbyte-faros-destination/resources/source-specific-configs/agileaccelerator.json#L5); could you clarify?
> > 3. in this iteration of the destination, is it still possible to have a single destination used by N different sources (i.e. fill several "source specific config" sections on the same destination)? Is it a problem for the current setup of CE?
> 
> 1. Yes
> 2. Every object defined in a oneOf has to have a const property, even if it's the only object, to be rendered properly in the Airbyte UI. The purpose of making it a `oneOf` at all is to have each source's config options in their own block in the Airbyte UI. Otherwise, all the configs would be displayed at the same "level" in the UI and it'd be impossible to determine which config applies to which source, plus ordering would be a mess.
> 3. Yes you can still configure multiple source configs.
>    image

You are not answering why, as far as I can tell, there are 2 nested oneOf: one in spec.json, and [one in each source config](https://github.com/faros-ai/airbyte-connectors/blob/25dbdb386183bf491915a7f8f000e69662c1a9ee/destinations/airbyte-faros-destination/resources/source-specific-configs/agileaccelerator.json#L5).

**Pull request merged**